### ISSUE TITLE
Local Storage is shared between profiles

### ISSUE THREAD

**Issue by Ygrex** [aidanholm/luakit#107]
steps to reproduce:

1. purge any existing local storage:

```
$ rm -rf ~/.local/share/webkitgtk/localstorage
```

2. run luakit with a new profile that gets created on the fly and quit it as soon as the page is loaded:

```
$ ./luakit -Up new https://github.com
```

3. check https_github.com_0.localstorage created in localstorage/, modify it directly:

```
$ sqlite3 ~/.local/share/webkitgtk/localstorage/https_github.com_0.localstorage "delete from Itemtable where key like 'logged-in'"
```

4. run luakit with one more new profile and quit it as above:

```
$ ./luakit -Up fresh https://github.com
```

5. check the same file in local storage is updated:

```
$ sqlite3 ~/.local/share/webkitgtk/localstorage/https_github.com_0.localstorage 'select * from Itemtable'
bundle-urls|{
logged-in|f
```

my expectations might be wrong but local storage should be scoped to a profile

**Comment by aidanholm** [comment]
Your expectations are correct; this is a bug. WebKit 2 has a WebKitWebsiteDataManager object which should be instantiated with the correct paths.

**Comment by aidanholm** [comment]
Fixed, as of 45ec30f.
